casper/src/main/resources/MakeMint.rho

```diff
@@ -31,18 +31,25 @@ in {
         } |
         contract internalMakePurse(balance, return) = {
           //balance must be a name that NonNegativeNumber contract is listening on
-          new thisPurse in {
+          new thisPurse, logStore in {
+            logStore!(Nil) |
             new letCh in { // let thisDecr = ...
               letCh!(bundle0{bundle0{*thisPurse}|*decr}) | for (thisDecr <- letCh) {
                 contract thisPurse(@=*thisDecr, @amount, return) = {
-                  new success in {
-                    balance!("sub", amount, *success) | for (@subOk <- success) {
-                      return!(subOk, *decr)
+                  for (logCh <<- logStore) {
+                    new success in {
+                      balance!("sub", amount, *success) | for (@subOk <- success) {
+                        if (Nil != *logCh) { logCh!(["*decr", amount, subOk]) } |
+                        return!(subOk, *decr)
+                      }
                     }
                   }
                 }
               }
             } |
+            contract thisPurse(@"setLog", logCh, ack) = {
+              for (_ <- logStore) { logStore!(*logCh) | ack!(Nil) }
```

Comment:
What is correct way to disable logging? Maybe it would be useful as comment here.

Is it makes sense to have method to check if log exists?

Reply:
> What is correct way to disable logging? Maybe it would be useful as comment here.

Set it to Nil.  I'll add a comment.

> Is it makes sense to have method to check if log exists?

It always exists because I set it on line 35 and then only peek for it/set it to something new.